expression/expression.go

```diff
@@ -1111,7 +1111,11 @@ func scalarExprSupportedByFlash(function *ScalarFunction) bool {
 // TODO: remove this function. ScalarFunction should be newed in one place.
 func wrapWithIsTrue(ctx sessionctx.Context, keepNull bool, arg Expression) (Expression, error) {
 	if arg.GetType().EvalType() == types.ETInt {
-		return arg, nil
+		if child, ok := arg.(*ScalarFunction); ok {
```

Comment:
Shall we add a context flag to indicate that we are in `PushDownNot` and then do this check? then we can avoid incurring unnecessary `IsTrue()` functions like those in the above explain test result.